Sources/PubNubChat/Storage/CoreData/CoreDataMigrationManager.swift

```diff
@@ -63,20 +64,45 @@ class DefaultCoreDataMigrationManager: CoreDataMigrationManager {
       return
     }
     
-    for modelVersion in nextModelVersions {
+    let allVersionFiles = modelBundle.paths(
+      forResourcesOfType: "mom",
+      inDirectory: modelURL.lastPathComponent
+    )
+    
+    let sortedManagedObjectModels = allVersionFiles.compactMap() {
+      NSManagedObjectModel(contentsOf: URL(fileURLWithPath: $0))
+    }.sorted() {
+      $0.integerValue < $1.integerValue
+    }
+    
+    let currentStoreMetadata = try NSPersistentStoreCoordinator.metadataForPersistentStore(
+      ofType: NSSQLiteStoreType,
+      at: persistentStoreLocation
+    )
+    
+    let identifiers = (currentStoreMetadata["NSStoreModelVersionIdentifiers"] as? [String] ?? []).first ?? String()
+    let integerVersionFromTheStore = Int(identifiers) ?? 0
+    let modelVersionMatchingStore = sortedManagedObjectModels.first(where: { $0.integerValue == integerVersionFromTheStore })!
+    
+    var currentlyProcessedModel = modelVersionMatchingStore
+    
+    for modelVersion in sortedManagedObjectModels.filter({ $0.integerValue > integerVersionFromTheStore }) {
```

Comment:
I would say that we were forcing to do a sequential migration. What's not working for us is the custom [entity migration policy](https://developer.apple.com/documentation/coredata/nsentitymigrationpolicyl) which doesn't want to fire. We have one migration policy so far (`PayloadAlignmentMessageEntityMigration`) that is responsible for retrieving a message text from the `content` property and put it into newly added `text` property.

I would love to create `NSMappingModel` from the Xcode instead of doing it in Swift, edit them if needed and let CoreData to perform these migrations for us. Apple [says](https://developer.apple.com/documentation/coredata/nsentitymigrationpolicy) in the Overview section that typically, you specify the name for the migration policy in the Xcode mapping model editor. I tried to set different values but nothing worked. I suspect that's because of `PubNubChatModel` that's located in different bundle than the application's bundle so it might work in a different way.

Reply:
I had the same assumption, so I moved the entire DB + Mapping Model into the application bundle, but was getting similar errors from when it was all embedded inside the framework.  It seems that Mapping Model files (like a lot of Core Data) is very poorly documented :(